substrate/primitives/core/src/bls.rs

```diff
@@ -238,183 +238,194 @@ mod tests {
-	);
-		let signature = Signature::unchecked_from(signature);
-		assert!(pair.sign(&message[..]) == signature);
-		assert!(Pair::verify(&signature, &message[..], &public));
-	}
-
-	#[test]
-	fn test_vector_by_string_should_work() {
-		let pair = Pair::from_string(
-			"0x9d61b19deffd5a60ba844af492ec2cc44449c5697b326919703bac031cae7f60",
-			None,
-		)
-		.unwrap();
-		let public = pair.public();
-		assert_eq!(
-			public,
-			Public::unchecked_from(array_bytes::hex2array_unchecked(
-				"7a84ca8ce4c37c93c95ecee6a3c0c9a7b9c225093cf2f12dc4f69cbfb847ef9424a18f5755d5a742247d386ff2aabb806bcf160eff31293ea9616976628f77266c8a8cc1d8753be04197bd6cdd8c5c87a148f782c4c1568d599b48833fd539001e580cff64bbc71850605433fcd051f3afc3b74819786f815ffb5272030a8d03e5df61e6183f8fd8ea85f26defa83400"
-			))
-		);
-		let message = b"";
-		let signature =
-	array_bytes::hex2array_unchecked("d1e3013161991e142d8751017d4996209c2ff8a9ee160f373733eda3b4b785ba6edce9f45f87104bbe07aa6aa6eb2780aa705efb2c13d3b317d6409d159d23bdc7cdd5c2a832d1551cf49d811d49c901495e527dbd532e3a462335ce2686009104aba7bc11c5b22be78f3198d2727a0b"
-	);
-		let expected_signature = Signature::unchecked_from(signature);
-		println!("signature is {:?}", pair.sign(&message[..]));
-		let signature = pair.sign(&message[..]);
-		assert!(signature == expected_signature);
-		assert!(Pair::verify(&signature, &message[..], &public));
-	}
-	#[test]
-	fn generated_pair_should_work() {
-		let (pair, _) = Pair::generate();
-		let public = pair.public();
-		let message = b"Something important";
-		let signature = pair.sign(&message[..]);
-		assert!(Pair::verify(&signature, &message[..], &public));
-		assert!(!Pair::verify(&signature, b"Something else", &public));
-	}
-
-	#[test]
-	fn seeded_pair_should_work() {
-		let pair = Pair::from_seed(b"12345678901234567890123456789012");
-		let public = pair.public();
-		assert_eq!(
-			public,
-			Public::unchecked_from(
-				array_bytes::hex2array_unchecked(
-				"754d2f2bbfa67df54d7e0e951979a18a1e0f45948857752cc2bac6bbb0b1d05e8e48bcc453920bf0c4bbd5993212480112a1fb433f04d74af0a8b700d93dc957ab3207f8d071e948f5aca1a7632c00bdf6d06be05b43e2e6216dccc8a5d55a0071cb2313cfd60b7e9114619cd17c06843b352f0b607a99122f6651df8f02e1ad3697bd208e62af047ddd7b942ba80080")
-			)
-		);
-		let message =
-	array_bytes::hex2bytes_unchecked("2f8c6129d816cf51c374bc7f08c3e63ed156cf78aefb4a6550d97b87997977ee00000000000000000200d75a980182b10ab7d54bfed3c964073a0ee172f3daa62325af021a68f707511a4500000000000000"
-	);
-		let signature = pair.sign(&message[..]);
-		println!("Correct signature: {:?}", signature);
-		assert!(Pair::verify(&signature, &message[..], &public));
-		assert!(!Pair::verify(&signature, "Other message", &public));
-	}
-
-	#[test]
-	fn generate_with_phrase_recovery_possible() {
-		let (pair1, phrase, _) = Pair::generate_with_phrase(None);
-		let (pair2, _) = Pair::from_phrase(&phrase, None).unwrap();
-
-		assert_eq!(pair1.public(), pair2.public());
+        fn default_phrase_should_be_used_for_bls377() {
```

Comment:
So if I understood correctly you want to share test code between bls12-381 and 377 right?
Looks fine